From 086b798c0d4ce17a409a04988910242868035f89 Mon Sep 17 00:00:00 2001
From: KrishnaSimmadhari <91877513+KrishnaSimmadhari@users.noreply.github.com>
Date: Wed, 30 Aug 2023 22:49:03 +0530
Subject: [PATCH] mfd: intel-m10-bmc: PMCI and log changes for CMC

Change register map for CMC card to c6100 register csr set. Change the FPGA info
structure to point to the right location.

Signed-off-by: Krishna Kumar S R <simmadhari.ramadass.krishna.kumar@intel.com>
---
 drivers/mfd/intel-m10-bmc-log.c  | 9 +++++++++
 drivers/mfd/intel-m10-bmc-pmci.c | 4 ++--
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/drivers/mfd/intel-m10-bmc-log.c b/drivers/mfd/intel-m10-bmc-log.c
index 12ef1100da79..0c747ba0d0bc 100644
--- a/drivers/mfd/intel-m10-bmc-log.c
+++ b/drivers/mfd/intel-m10-bmc-log.c
@@ -245,6 +245,11 @@ static const struct m10bmc_log_cfg m10bmc_log_c6100_cfg = {
 	.id_off = M10BMC_C6100_FPGA_IMAGE_DIR_ADDR,
 };
 
+static const struct m10bmc_log_cfg m10bmc_log_cmc_cfg = {
+	.id_size = M10BMC_C6100_FPGA_IMAGE_DIR_SIZE,
+	.id_off = M10BMC_C6100_FPGA_IMAGE_DIR_ADDR,
+};
+
 static const struct platform_device_id intel_m10bmc_log_ids[] = {
 	{
 		.name = "n6000bmc-log",
@@ -254,6 +259,10 @@ static const struct platform_device_id intel_m10bmc_log_ids[] = {
 		.name = "c6100bmc-log",
 		.driver_data = (unsigned long)&m10bmc_log_c6100_cfg,
 	},
+	{
+		.name = "cmcbmc-log",
+		.driver_data = (unsigned long)&m10bmc_log_cmc_cfg,
+	},
 	{ }
 };
 
diff --git a/drivers/mfd/intel-m10-bmc-pmci.c b/drivers/mfd/intel-m10-bmc-pmci.c
index f5042042a8aa..35e880bec80d 100644
--- a/drivers/mfd/intel-m10-bmc-pmci.c
+++ b/drivers/mfd/intel-m10-bmc-pmci.c
@@ -362,7 +362,7 @@ static struct mfd_cell m10bmc_pmci_c6100_bmc_subdevs[] = {
 static struct mfd_cell m10bmc_pmci_cmc_bmc_subdevs[] = {
 	{ .name = "cmcbmc-hwmon" },
 	{ .name = "n6000bmc-sec-update" },
-	{ .name = "n6000bmc-log" },
+	{ .name = "cmcbmc-log" },
 };
 
 static const struct m10bmc_csr_map m10bmc_n6000_csr_map = {
@@ -430,7 +430,7 @@ static const struct intel_m10bmc_platform_info m10bmc_pmci_c6100 = {
 static const struct intel_m10bmc_platform_info m10bmc_pmci_cmc = {
 	.cells = m10bmc_pmci_cmc_bmc_subdevs,
 	.n_cells = ARRAY_SIZE(m10bmc_pmci_cmc_bmc_subdevs),
-	.csr_map = &m10bmc_n6000_csr_map,
+	.csr_map = &m10bmc_c6100_csr_map,
 };
 
 static int m10bmc_pmci_probe(struct dfl_device *ddev)
